cmd/graphql2go/main.go

```diff
@@ -358,6 +379,53 @@ func newGenerator(outWriter io.Writer, root *ast.Document) *generator {
 	return g
 }
 
+func (g *generator) assertAllowIdentityAssumptionConditions(root *ast.Document) {
+	for _, def := range root.Definitions {
+		switch d := def.(type) {
+		case *ast.ObjectDefinition:
+			if isTopLevelObject(d.Name.Value) {
+				g.assertAllowIdentityAssumptionConditionsOnFields(d.Name.Value, d.Fields, make(map[string]struct{}), true)
+			}
+		}
+	}
+}
+
+func (g *generator) assertAllowIdentityAssumptionConditionsOnFields(parentName string, fieldDefs []*ast.FieldDefinition, seenTypes map[string]struct{}, requiredOnAll bool) {
+	for _, f := range fieldDefs {
+		allow, ok := identityAssumptionDirectiveAllowValue(f.Directives)
+		if !ok && requiredOnAll {
+			log.Fatalf("@allowAssumedIdentity directive required on field %q since parent %q is top level or has @allowAssumedIdentity(allow: true)", f.Name.Value, parentName)
```

Comment:
```suggestion
			g.failf("@allowAssumedIdentity directive required on field %q since parent %q is top level or has @allowAssumedIdentity(allow: true)", f.Name.Value, parentName)
```